lib/JSON/Validator.pm

```diff
@@ -586,13 +586,15 @@ sub _validate_all_of {
 
   return if not @errors;
 
+  my @details = map { map $_->details, @{$_}[1 .. $#{$_}] } @errors;
+
   return prefix_errors(allOf => @errors)
-    if @errors == 1
-    or grep $_->[1]->details->[1] ne 'type', @errors;
+    if @details == 1
+    or grep $_->[1] ne 'type', @details;
```

Comment:
I also find grep without {} hard to read, especially when it's not just a simple expression.

`$_->[1]` == simple
`$_->[1] ne 'type'` != simple